### PR TITLE
core: exit with failure if suite doesn't start

### PR DESCRIPTION
**Pull request by rcooke-warwick**
resolves https://github.com/balena-os/leviathan/issues/811

When a device type was passed to the suite that wasn't in the publicly available device types contracts, the suite would fail in the constructor. The error was never caught and the suite subprocess would exit with a zero, leading to the suite passing. 

I added a try/catch to catch the error and exit with a failure if it happens

Change-type: patch
Signed-off-by: Ryan Cooke <ryan@balena.io>

### CODE REVIEW

File: core/lib/common/suiteSubprocess.js [hunk 1]
```diff
@@ -63,56 +63,62 @@ async function createJsonSummary(suite) {
...
+		suite.state.log(`Suite run complete`);
+		process.off('message', messageHandler);
+		suite.state.log(`Exiting suite child process...`);
+		if (suite.passing) {
+			process.exit();
+		} else {
+			process.exit(1);
+		}
+    }catch(e){
+		process.exitCode = 4;
```
Comment (klutchell) [inline core/lib/common/suiteSubprocess.js:119]: Looks like this file needs to be linted, but looks good other than that.

**Review by vipulgupta2048 — COMMENTED** [review]
Thanks @rcooke-warwick for taking this up. I had a different solution in mind which might be more elegant. Instead of approaching this from the core side, I think we put should put the any test run on the client side as  `1` or `failed` by default instead of default success. That way we won't need to emit an error here to then be caught on client side in the end. 

We always consider failure then I think that's a better scenario and possibly even be an improvement beyond this patch.

**Comment by rcooke-warwick** [comment]
@vipulgupta2048 at first that what I also thought ^ however even if the default state of the client is "failure", if there is an error in the constructor of the suite, the suite subprocess will exit and provide the "pass" signal to the client. This meant I had to do it this way instead

**Review by vipulgupta2048 — APPROVED** [review]
Right on right on

**Pull request merged**